spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/SpringDocWebSecurityCustomizer.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.config;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityCustomizer;
+
+/**
+ * Makes SpringDoc public available without any authentication required.
+ *
+ * @author Tobias Soloschenko
+ */
+@Configuration
```

Comment:
I think we may want to name this more generic such as `SpringDocConfiguration` as we will want to put the FilterRegistrationBean in it as well (probably). Instead of `implements WebSecurityCustomizer` we can create a bean method that returns a lambda for the customizer such as:
```java
    @Bean
    public WebSecurityCustomizer springDocCustomizer() { 
        return (webSecurity -> {
            ....
        });
    }
```

Reply:
Done